Makefile

```diff
@@ -35,12 +35,16 @@ example_usage_sht3x_hw: ${common_objects} ${hw_objects} sht3x.o
 	$(CC) $(LDFLAGS) -o $@ $^ $(LOADLIBES) $(LDLIBS)
 
 release:
-	./release.py
+	export tag="$$(git describe --always --dirty)" && \
+	./release.py && \
 	for directory in release/*; do \
 		echo $${directory}; \
 		cd $${directory}; \
-		$(CC) $(LDFLAGS) -o example_usage_$$(basename $${directory}) *.c; \
-		$(RM) example_usage_$$(basename $${directory}); \
+		export subdir="$$(basename $${directory})"; \
+		export drv=$${subdir%_*_i2c}; \
+		perl -pi -e "s/^(#define\s+SHT_DRV_VERSION_STR\s+)\".*\"$$/\\1\"$${tag}\"/" "$${drv}.h"; \
+		$(CC) $(LDFLAGS) -o example_usage_$${subdir} *.c; \
+		$(RM) example_usage_$${subdir}; \
```

Comment:
Why not generate a file with git describe in the Makefile and include it in sht.h? Would save the hassle of having to replace it with a perl call.
The file could be generated on every build.